### ISSUE TITLE
Unknown 'bem' function

### ISSUE THREAD

**Issue by apetkevicius** [fourkitchens/emulsify#234]
Hi guys! I have a problem using emulsify subtheme. I installed theme and all dependencies following instructions in the readme file. I commited everything in a git repo. Other developer pulled and he got this error (while trying to load a page):
`The website encountered an unexpected error. Please try again later.</br></br><em class="placeholder">Twig_Error_Syntax</em>: Unknown &quot;bem&quot; function. in <em class="placeholder">Twig_ExpressionParser-&gt;getFunctionNodeClass()</em> (line <em class="placeholder">21</em> of <em class="placeholder">themes/custom/custom_theme/components/_patterns/01-atoms/04-images/icons/_icon.twig</em>). <pre class="backtrace">Twig_ExpressionParser-&gt;getFunctionNode(&#039;bem&#039;, 21) (Line: 162)`

I found out the problem is that unknown `bem` function file `/components/_twig-components/functions/bem.function.php` is in the `.gitignore`. But I can't understand why.

Maybe someone faced the same problem?

**Comment by evanmwillhite** [comment]
On a high level, all dependencies installed via composer and npm/yarn are gitignored, so your other developer will also need to run through the install steps locally, which when done manually are in this order:

1. `composer install` 
2. `npm install` or `yarn install`

This will make sure everything is installed correctly, including the bem twig extension. We host the BEM function [separately](https://github.com/drupal-pattern-lab/bem-twig-extension) as a part of the Drupal Pattern Lab initiative since it could be used outside Emulsify. So it is installed as a dependency and moved into the correct location in a post-install script.

**Comment by apetkevicius** [comment]
Thank you for an answer! But what if I want to deploy theme to production? Is it a good practice that we need to run `yarn install` there? Then we would need to keep a lot of not necessary files like node_modules folder, pattern-lab folder and so on.

Have you any idea how to solve it?

**Comment by ModulesUnraveled** [comment]
Answered in slack, but wanted to add this here for posterity, and future potential searchers:

>Just wanted to chime in, part of the Emulsify setup (I think part of the composer post-install script) moves the bem and addattributes functions into `<theme>/components/_twig-components/functions` so you can safely gitignore the node_modules directory. But if you don’t have your build tool adding the functions folder to your repo, you should disable the default ignore rules. (we use CircleCI to build an “artifact” repo that gets pushed to Pantheon. That repo includes the functions folder)

**Comment by miller2676** [comment]
@ModulesUnraveled -- Just to be clear, are you saying that we should remove `**/bem.function.php` from the gitignore for deployment to production? Anything else?

**Comment by ModulesUnraveled** [comment]
Yes, or force add it to your repo. Either way, you will need the bem function on your production server.

You shouldn't need to make any other changes, unless you're using the `add_attributes` function. (It's not used by default). If you do use that, you'll need to remove that line as well.

Note: I just noticed the bem function was in the gitignore file twice, so I've created a PR to remove one of them (https://github.com/fourkitchens/emulsify/pull/241)
